tests/samplers_tests/test_partial_fixed.py

```diff
@@ -0,0 +1,95 @@
+        x = trial.suggest_uniform("x", -10, 10)
+        y = trial.suggest_uniform("y", -10, 10)
+        return x ** 2 + y ** 2
+
+    study0 = optuna.create_study()
+    study0.sampler = RandomSampler(seed=42)
+    study0.optimize(objective, n_trials=1)
+    x_sampled0 = study0.trials[0].params["x"]
+
+    # Fix parameter ``y`` as 0.
+    study1 = optuna.create_study()
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        study1.sampler = PartialFixedSampler(
+            fixed_params={"y": 0}, base_sampler=RandomSampler(seed=42)
+        )
+    study1.optimize(objective, n_trials=1)
+    x_sampled1 = study1.trials[0].params["x"]
+    y_sampled1 = study1.trials[0].params["y"]
+    assert x_sampled1 == x_sampled0
+    assert y_sampled1 == 0
+
+
+@pytest.mark.parametrize("fixed_y", [0.5, 5])
+def test_incompatible_types(fixed_y: Union[float, int]) -> None:
+    def objective(trial: Trial) -> float:
+        x = trial.suggest_int("x", -10, 10)
+        y = trial.suggest_int("y", -10, 10)
+        return x ** 2 + y ** 2
+
+    # Parameters of Int-type-distribution are rounded to int-type,
+    # even if they are specified as float-type.
+    study = optuna.create_study()
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        study.sampler = PartialFixedSampler(
+            fixed_params={"y": fixed_y}, base_sampler=study.sampler
+        )
+    study.optimize(objective, n_trials=1)
+    assert study.trials[0].params["y"] == int(fixed_y)
+
+
+@pytest.mark.parametrize("fixed_y", [-2, 2])
+def test_out_of_the_range(fixed_y: int) -> None:
+    def objective(trial: Trial) -> float:
+        x = trial.suggest_int("x", -1, 1)
+        y = trial.suggest_int("y", -1, 1)
+        return x ** 2 + y ** 2
+
+    # It is possible to fix parameters as out-of-the-range value.
+    # Userwarnings will occur.
+    study = optuna.create_study()
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        study.sampler = PartialFixedSampler(
+            fixed_params={"y": fixed_y}, base_sampler=study.sampler
+        )
+    with pytest.warns(UserWarning):
+        study.optimize(objective, n_trials=1)
+    assert study.trials[0].params["y"] == fixed_y
+
+
+def test_partial_fixed_experimental_warning() -> None:
+    study = optuna.create_study()
+    with pytest.warns(optuna.exceptions.ExperimentalWarning):
+        optuna.samplers.PartialFixedSampler(fixed_params={"x": 0}, base_sampler=study.sampler)
+
+
+def test_reseed_rng() -> None:
+    study = optuna.create_study()
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = PartialFixedSampler(fixed_params={"x": 0}, base_sampler=study.sampler)
+    original_seed = sampler._base_sampler._rng.seed  # type: ignore
```

Comment:
I added `# type: ignore` because `mypy` fails.
I don't know the reason for this...If anyone knows why, please let me know.

Reply:
On the original codebase, I thought the reason was that the parent class, `BaseSampler`, did not have `seed` and `_rng` variables, so mypy showed the error. The mypy error could not be fixed by removing the assignment of `reseed_rng`. I'm sorry if I confuse you.

I think `FixedSampler` does not need to store `_rng` as in https://github.com/optuna/optuna/pull/1892#discussion_r515696516, so it is supposed to be okay to add `  # type: ignore` in the test to store the original seed and comparison, but I'm not sure. 

Anyway, we have time to discuss this PR online tomorrow with @HideakiImamura . Let's continue doing this PR then.

Reply:
Thanks at all :)